x-pack/plugins/upgrade_assistant/server/lib/es_version_precheck.ts

```diff
@@ -62,11 +62,11 @@ export const esVersionCheck = async (
   ctx: RequestHandlerContext,
   response: KibanaResponseFactory
 ) => {
-  const { adminClient } = ctx.core.elasticsearch;
+  const { client } = ctx.core.elasticsearch.legacy;
```

Comment:
I see that this used to consume `adminClient`. What was the difference between `adminClient` and `dataClient`? Why are we able to replace both of them with `legacy.client`?

Reply:
```
For "tribe mode" support, we used to expose two elasticsearch clients a "data" and "admin" client. Since Kibana no longer supports "tribe mode" Core should expose a single Elasticsearch client.
```
from 
https://github.com/elastic/kibana/issues/49870